templates/repo/home.tmpl

```diff
@@ -4,171 +4,222 @@
+						{{if $isHomepage}}
+							<div class="clone-panel ui action tiny input">
+								{{template "repo/clone_buttons" .}}
+								<button class="ui small jump dropdown icon button" data-tooltip-content="{{ctx.Locale.Tr "repo.more_operations"}}">
+									{{svg "octicon-kebab-horizontal"}}
+									<div class="menu">
+										{{if not $.DisableDownloadSourceArchives}}
+											<a class="item archive-link" href="{{$.RepoLink}}/archive/{{PathEscapeSegments $.RefName}}.zip" rel="nofollow">{{svg "octicon-file-zip" 16 "tw-mr-2"}}{{ctx.Locale.Tr "repo.download_zip"}}</a>
+											<a class="item archive-link" href="{{$.RepoLink}}/archive/{{PathEscapeSegments $.RefName}}.tar.gz" rel="nofollow">{{svg "octicon-file-zip" 16 "tw-mr-2"}}{{ctx.Locale.Tr "repo.download_tar"}}</a>
+											<a class="item archive-link" href="{{$.RepoLink}}/archive/{{PathEscapeSegments $.RefName}}.bundle" rel="nofollow">{{svg "octicon-package" 16 "tw-mr-2"}}{{ctx.Locale.Tr "repo.download_bundle"}}</a>
+										{{end}}
+										{{range .OpenWithEditorApps}}
+											<a class="item js-clone-url-editor" data-href-template="{{.OpenURL}}">{{.IconHTML}}{{ctx.Locale.Tr "repo.open_with_editor" .DisplayName}}</a>
+										{{end}}
+									</div>
+								</button>
+								{{template "repo/clone_script" .}}{{/* the script will update `.js-clone-url` and related elements */}}
+							</div>
+							{{template "repo/cite/cite_modal" .}}
+						{{end}}
+						{{if and (not $isHomepage) (not .IsViewFile) (not .IsBlame)}}{{/* IsViewDirectory (not home), TODO: split the templates, avoid using "if" tricks */}}
+							<a class="ui button" href="{{.RepoLink}}/commits/{{.BranchNameSubURL}}/{{.TreePath | PathEscapeSegments}}">
+								{{svg "octicon-history" 16 "tw-mr-2"}}{{ctx.Locale.Tr "repo.file_history"}}
 							</a>
-						</div>
-					</button>
+						{{end}}
+					</div>
+				</div>
+				{{if .IsViewFile}}
+					{{template "repo/view_file" .}}
+				{{else if .IsBlame}}
+					{{template "repo/blame" .}}
+				{{else}}{{/* IsViewDirectory */}}
+					{{template "repo/view_list" .}}
 				{{end}}
+			</div>
 
-				{{if and $isHomepage (.Repository.IsTemplate)}}
-					<a role="button" class="ui primary compact button" href="{{AppSubUrl}}/repo/create?template_id={{.Repository.ID}}">
-						{{ctx.Locale.Tr "repo.use_template"}}
-					</a>
-				{{end}}
-				{{if $isHomepage}}
-					{{/* only show the "code search" on the repo home page, it only does global search,
-						so do not show it when viewing file or directory to avoid misleading users (it doesn't search in a directory) */}}
-					<form class="ignore-dirty tw-flex tw-flex-1" action="{{.RepoLink}}/search" method="get">
-						<div class="ui small action input tw-flex-1">
-							<input name="q" size="10" placeholder="{{ctx.Locale.Tr "search.code_kind"}}">
-							{{template "shared/search/button"}}
+			{{if $isHomepage}}
+				<div class="flex-container-repo">
```

Comment:
Do not understand why the elements must be duplicated.

Reply:
![image](https://github.com/user-attachments/assets/bc313e5a-98e5-4d94-b27e-80611f9b957b)
In mobile view, part 2 will be displayed at first:
![image](https://github.com/user-attachments/assets/9b128e32-6702-4b46-9939-1e327f151347)
![image](https://github.com/user-attachments/assets/608b6f19-ef59-4408-9fd9-77c9d151e468)
Is there a way to adjust the location of part 2 in the first picture without using two same part?

Reply:
google flex order

Reply:
flex order can move these parts, but it seems that it is hard to implement the first pic, as flex can not have two directions in the same container as the same time.
I will Mark this PR to draft.

Reply:
There is also "grid order"

https://jsfiddle.net/s5qjgf7d/

Reply:
Done.